core/src/main/kotlin/com/malinskiy/marathon/execution/strategy/impl/flakiness/ProbabilityBasedFlakinessStrategy.kt

```diff
@@ -0,0 +1,38 @@
+package com.malinskiy.marathon.execution.strategy.impl.flakiness
+
+import com.malinskiy.marathon.analytics.metrics.MetricsProvider
+import com.malinskiy.marathon.execution.TestShard
+import com.malinskiy.marathon.execution.strategy.FlakinessStrategy
+import com.malinskiy.marathon.test.Test
+import java.time.Instant
+
+/**
+ * The idea is that flakiness anticipates the flakiness of the test based on the probability of test passing
+ * and tries to maximize the probability of passing when executed multiple times.
+ * For example the probability of test A passing is 0.5 and configuration has probability of 0.8 requested,
+ * then the flakiness strategy multiplies the test A to be executed 3 times
+ * (0.5 x 0.5 x 0.5 = 0.125 is the probability of all tests failing, so with probability 0.875 > 0.8 at least one of tests will pass).
+ */
+
+class ProbabilityBasedFlakinessStrategy(private val minSuccessRate: Double,
```

Comment:
Should we place some thresholds here? If the probability is very low then maybe we don't want to add the same test 1000 times?

Reply:
added maxCount param.